### PR TITLE
Updated test_encore.py

### LINKED ISSUE #3695: tests should use results.rmsf to avoid DeprecationWarning
## Expected behavior ##

Our tests do not use deprecated constructs that will be removed. 

## Actual behavior ##

Tests trigger `DeprecationWarning` for uses of using the `rmsf` attribute instead of `results.rmsf`
```
analysis/test_encore.py::TestEncore::test_ensemble_superimposition
analysis/test_encore.py::TestEncore::test_ensemble_superimposition
analysis/test_encore.py::TestEncore::test_ensemble_superimposition_to_reference_non_weighted
analysis/test_encore.py::TestEncore::test_ensemble_superimposition_to_reference_non_weighted
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/MDAnalysis/analysis/rms.py:885: DeprecationWarning: The `rmsf` attribute was deprecated in MDAnalysis 2.0.0 and will be removed in MDAnalysis 3.0.0. Please use `results.rmsd` instead.
    warnings.warn(wmsg, DeprecationWarning)

analysis/test_encore.py::TestEncore::test_rmsd_matrix_with_superimposition
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/MDAnalysis/analysis/rms.py:718: DeprecationWarning: The `rmsd` attribute was deprecated in MDAnalysis 2.0.0 and will be removed in MDAnalysis 3.0.0. Please use `results.rmsd` instead.
    warnings.warn(wmsg, DeprecationWarning)
```

## Code to reproduce the behavior ##

Run the tests listed above.

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 2.2.0
- Which version of Python (`python -V`)? 3.8
- Which operating system? Linux

### PR DESCRIPTION
**Pull request by v-parmar**
Fixed #3695

Updated test_encore.py file.
Updated 'rmsd' attribute with 'results.rmsd' and 'rmsf' with 'results.rmsf' in test_encore.py file.

### CODE REVIEW

File: testsuite/MDAnalysisTests/analysis/test_encore.py [hunk 1]
```diff
@@ -211,8 +211,8 @@ def test_ensemble_superimposition(self):
         rmsfs2 = rms.RMSF(aligned_ensemble2.select_atoms('name *'))
         rmsfs2.run()
 
-        assert sum(rmsfs1.rmsf) > sum(rmsfs2.rmsf),"Ensemble aligned on all " \
-                                                   "atoms should have lower full-atom RMSF than ensemble aligned on only CAs."
+        assert sum(rmsfs1.results.rmsf) > sum(rmsfs2.results.rmsf), "Ensemble aligned on all " \
+                                                                    "atoms should have lower full-atom RMSF than ensemble aligned on only CAs."
```
Comment (RMeli) [inline testsuite/MDAnalysisTests/analysis/test_encore.py]: These lines (and the ones above) do not satisfy MDAnalysis linters. You can see that the CI is failing and find more information here: https://github.com/MDAnalysis/mdanalysis/actions/runs/4253218551/jobs/7397788525

**Review by RMeli — CHANGES_REQUESTED** [review]
Thank you for the PR @v-parmar. As you can see, the "linters" checks are not successful. You need to format the code properly in order for them to pass.

Also, since you are a first time contributor, you need to add your name to the `package/AUTHORS` file, as well as your GitHub handle to `package/CHANGELOG`.

Finally, this PR solves the issue for `test_encore.py`. Have you checked if the same happens for other tests, as suggested by @orbeckst in https://github.com/MDAnalysis/mdanalysis/issues/3695#issuecomment-1424334769?

File: testsuite/MDAnalysisTests/analysis/test_encore.py (hunk 1 above)
Comment (v-parmar) [reply testsuite/MDAnalysisTests/analysis/test_encore.py]: As you said I have done all the changes.

**Comment by RMeli** [comment]
Thanks @v-parmar. There are now conflicts in `package/CHANGELOG` that need to be solved.

> Finally, this PR solves the issue for test_encore.py. Have you checked if the same happens for other tests, as suggested by @orbeckst in https://github.com/MDAnalysis/mdanalysis/issues/3695#issuecomment-1424334769?

Can you please confirm if you have checked this?

**Comment by v-parmar** [comment]
> Thanks @v-parmar. There are now conflicts in `package/CHANGELOG` that need to be solved.
> 
> > Finally, this PR solves the issue for test_encore.py. Have you checked if the same happens for other tests, as suggested by @orbeckst in [#3695 (comment)](https://github.com/MDAnalysis/mdanalysis/issues/3695#issuecomment-1424334769)?
> 
> Can you please confirm if you have checked this?

Yes, I have checked.

**Comment by RMeli** [comment]
@v-parmar unfortunately the way you changed the `CHANGELOG` is not quite right. You need to add your GitHub handle alongside the handles of other developers, for the `2.5.0` release. The `??/??/??` is just a placeholder for the date of the release, which will be updated during the release process.

Please move your GitHub handle and the description of the change to the appropriate place. Only the description under "Fixes" is needed for this PR.

You will likely get back the conflicts you were having before, that you will need to fix. You can use the [GitHub Web interface](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github) to solve conflicts. See [addressing merge conflicts](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts) for more details.

File: package/CHANGELOG [hunk 2]
```diff
@@ -13,12 +13,14 @@ The rules for this file:
...
 ------------------------------------------------------------------------------
-
+  
 ??/??/?? IAlibay, pgbarletta, mglagolev, hmacdope, manuel.nuno.melo, chrispfae, 
-         ooprathamm, MeetB7, BFedder
+         ooprathamm, MeetB7, BFedder, v-parmar
  * 2.5.0
 
 Fixes
+
```
Comment (RMeli) [inline package/CHANGELOG]:
```suggestion
```

**Review by RMeli — APPROVED** [review]
Thanks @v-parmar , LGTM!

**Review by richardjgowers — APPROVED** [review]

**Pull request merged**